ccat.go

```diff
@@ -68,8 +69,18 @@ func CCat(fname string, p CCatPrinter, w io.Writer) error {
 		if err != nil {
 			return err
 		}
+
 		defer file.Close()
-		r = file
+
+		fi, err := file.Stat()
+
+		switch mode := fi.Mode(); {
+		case mode.IsDir():
+			fmt.Println(file.Name() + ": Is a directory")
+			os.Exit(1)
```

Comment:
We could return an error here and main.go will take care of exiting the program with the right error code: https://github.com/LadyDascalie/ccat/blob/09c185b4e142906c72de2448a30b40cd703f490d/main.go#L83.

```go
return fmt.Errorf("%s is a directory", file.Name())
```